### ISSUE TITLE
Freeze at Menu on AMD with ImGui

### ISSUE THREAD

**Issue by kilbouri** [SteamKaibz/NewColussus_Qols_Mod#3]
As noted in the Nexus description, when the ImGui is enabled (`"isUseImGui": true`) the game freezes immediately on loading to the main menu. I'm opening the issue to extend an offer of assistance in debugging this issue, as I have an RX 6750XT.

Obviously since this is a known issue, you may close this issue if you wish to not support AMD. Thanks for all your effort making this mod <3

**Comment by SteamKaibz** [comment]
Thank you kilbouri.
I  read your other issue reports. Actually...i'm in the process of completely rewriting another similar mod, for doom eternal atm with a collaborator, but as Doom Eternal has similarities with this mod, this should give a lot of insights as to what is going wrong with this mod, not only for the Imgui issue for AMD users but also for performance degradation (which seems fine however in New Colossus).
So there will be a rewrite of the New colussus mod as well, and if you're still interested later in giving feedback or even collaborating on the rewrite if you have c++ knowledge, let me know. 
But in any case it's going to take weeks/months, for all of this to be completed.

**Comment by kilbouri** [comment]
I do have some C++ experience. I wouldn't know where to start on a rewrite for the mod, but I am certainly open to helping develop and test the rewrite when you get started on it.

If you are open to having another contributor for the Doom Eternal mod, I'm happy to contribute there as well. (I'd need to buy the game to test it but I've been looking for an excuse for a while anyway lol)

**Comment by SteamKaibz** [comment]
Excellent, well i would suggest to wait for a good deal on Steam first. 
I'm still waiting for a life sign from my collaborator for Doom Eternal, but if nothing happens, i will switch back to New Colossus. Btw if you know anybody who has vulkan knowledge don't hesitate to tell them about this project.

**Comment by SteamKaibz** [comment]
Btw @kilbouri i don't know if you still have the game installed but i released a new version that may fix issues for AMD users. I also updated the repo that should now (hopefully) build just fine.

**Comment by kilbouri** [comment]
@SteamKaibz can confirm, ImGui now works on my RX 6750XT. Awesome! Reading the thread linked in the commit log, it was a semaphore deadlock and/or the missing frame dimensions?

**Comment by SteamKaibz** [comment]
Thanks a lot for the feedback ! 
I'll be honest with you, i do not know as i just implemented the suggested changes, tried it on my machine and published it as soon as i could so more users could potentially test it and finally enjoy the games/mods, but the programmer seems to suggest that the call to GetClientRect could be the key but there is a loooot of studying to do on my part...

**Comment by SteamKaibz** [comment]
Latest mod version does not use Imgui anymore

**Comment by kilbouri** [comment]
Can confirm the latest version is working on my GPU without issue. Moreover, in between when I opened this issue and now, I jumped over to Linux and I can also confirm that the mod works just fine via Proton. Just add the following to custom launch options (after following normal install procedure): `WINEDLLOVERRIDES="XINPUT1_3.dll=n" %command%` (the `=n` after the DLL name tells WINE to prefer the local copy of the DLL, rather than it's built-in version... use `n,b` instead of `n` if you want to fall back to the built-in one automatically).

Once again, huge huge thanks for making this mod - if I can buy you a coffee or something I absolutely will :)
